test/common/testlib.py

```diff
@@ -843,40 +844,41 @@ def check_issue(self, trace):
             traceback.print_exc()
         return number
 
-    def check_retry(self, trace):
-        #
+    def check_retry(self, trace, tries):
+
+        # Never try more than five times
+        if tries >= 5:
```

Comment:
It's a different thing.  The 5 in the retry loop is the hard limit to avoid infinite retries if check_return is buggy.  The 5 here is the real policy decision of how many times to actually retry individual failures.  I just put it to 5 for all cases, because it's anyway arbitrary.

We can get rid of the hard limit, and rely on the global timeout.